src/app/ui-helper/entity-subrecord/entity-subrecord.component.ts

```diff
@@ -140,4 +156,48 @@ export class EntitySubrecordComponent implements OnInit, OnChanges {
     col.selectValues = col.allSelectValues.filter(v => v.value.includes(input) || v.label.includes(input));
   }
 
+  // resets columnsToDisplay depending on current screensize
+  setupTable() {
+    if (this.columns !== undefined && this.screenWidth !== '') {
+      this.newColumns = [];
+      this.columns.forEach( (col) => this.checkVisibility(col));
+      this.columnsToDisplay = this.newColumns;
+      this.columnsToDisplay.push('actions');
+    }
+  }
+
+  // compares the current screensize to the columns' property visibleFrom. screensize < visibleFrom? column not displayed
+  checkVisibility(col) {
+    switch (col.visibileFrom) {
```

Comment:
couldn't we simplify this? Something like 
`
if (this.screenWidth.includes(col.name)) {
  this.newColumns.push(col.name);
}
`